src/algebra/module/zlattice.lean

```diff
@@ -0,0 +1,207 @@
+/-
+Copyright (c) 2023 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+import analysis.normed_space.basic
+import measure_theory.group.fundamental_domain
+import measure_theory.measure.haar_of_basis
+
+/-!
+# ℤ-lattices
+Let `E` be a finite dimensional real vector space. A (full) ℤ-lattice `L` of `E` is a discrete
+sub-ℤ-module of `E` such that `L` spans `E` over `ℝ`.
+
+The ℤ-lattice `L` can be defined in two ways:
+* For `b` a basis of `E`, then `L : submodule.span ℤ (set.range b)` is a ℤ-lattice of `E`.
+* As `L : submodule ℤ E` with the additional properties:
```

Comment:
Why not an `add_subgroup E`?

Reply:
(which is to say: either change it, or include an explanation of why the `submodule` spelling is more convenient)

Reply:
Yes, you are it should be `add_subgroup`. I had that in the original version and I changed at some point for some reason I don't recall.